deepchecks/vision/checks/data_integrity/abstract_property_outliers.py

```diff
@@ -215,47 +214,61 @@ def get_default_properties(self, data: VisionData):
         """Return default properties to run in the check."""
         pass
 
-    def _update_lowest_highest_property_values(self, images: t.List, labels: t.List, property_values: t.List,
-                                               property_name: str):
+    def _cache_property_values_and_images(self, images: t.List, labels: t.List, property_values: t.List,
+                                          property_name: str):
         """Update the _lowest_property_value_images, _lowest_property_value_images dicts based on new batch."""
+        is_property_per_label = isinstance(property_values[0], (np.ndarray, t.Sequence))
+        # Update full property values cache for outlier calculation
+        if is_property_per_label:
+            self._properties_results[property_name].extend(property_values)
+        else:
+            self._properties_results[property_name].extend([[x] for x in property_values])
```

Comment:
What is self._properties_results actually used for?

Reply:
the actual outlier calculation (stores all values then calculate based on them)